src/utils/GitLabUtils/GitLabUtils.test.js

```diff
@@ -0,0 +1,68 @@
+import { expect } from 'chai';
+import * as GitLabUtils from './GitLabUtils';
+import * as TestUtils from '../TestUtils/TestUtils';
+
+describe('GitLabUtils', () => {
+  describe('mergeCalendarsContributions', () => {
+    const actualCalendarFirstDate = '2019-04-18';
+    const contributionsOnFirstDate = 5;
+
+    const actualCalendar = TestUtils.getFakeContributionsObjectWithDailyCounts(
+      [contributionsOnFirstDate],
+      [actualCalendarFirstDate],
+    )[0];
```

Comment:
Cool improvement ideas, I'll take a deeper look into them asap.

Regarding the question in your comment, you're right that `mergeCalendarsContributions` works for the whole calendar(to be more precise, it should work), but I tried to adjust the tests to the returned fake calendar(s) - as `getFakeContributionsObjectWithDailyCounts` always returns a single day along with the passed contributions.

In general, the two different cases should test whether the function adds the right contributions to a particular date or not.

Reply:
> getFakeContributionsObjectWithDailyCounts always returns a single day

Why the `[0]` then?

Reply:
That happens because `map` returns an array - take a look at the implementation of `getFakeContributionsObjectWithDailyCounts `.

Reply:
But then why return an array if it only returns a single day and you have to get the first item anyway? My point is, can you make it so that they work on the same structure? I think the preferable structure is multiple days because that's how a real calendar looks like.

Reply:
It should be restructured, you're right.

Let me elaborate this in the subsequent PR as the changes will also affect other parts of tests.